### PR TITLE
Replace raw types with dynamic type arguments

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

File: built_value_generator/lib/src/serializer_source_field.dart [hunk 1]
```diff
@@ -193,12 +193,24 @@ abstract class SerializerSourceField
...
+    if (topLevel && DartTypes.isBuiltCollectionTypeName(bareType)) {
+      if (bareType == 'BuiltList' || bareType == 'BuiltSet') {
+        generics = 'dynamic';
+      } else if (bareType == 'BuiltMap' ||
+          bareType == 'BuiltListMultimap' ||
+          bareType == 'BuiltSetMultimap') {
+        generics = 'dynamic, dynamic';
+      } else {
+        // Should not get here; only 5 classes satisfy
+        // `isBuiltCollectionTypeName`.
```
Comment (srawlins) [inline built_value_generator/lib/src/serializer_source_field.dart]: WDYT for here? throw? throw Unsupported or something?

Comment (davidmorgan) [reply built_value_generator/lib/src/serializer_source_field.dart]: throw UnsupportedError SGTM.

Comment (srawlins) [reply built_value_generator/lib/src/serializer_source_field.dart]: Done

File: built_value_generator/lib/src/serializer_source_class.dart [hunk 2]
```diff
@@ -270,7 +270,7 @@ class _\$${name}Serializer implements StructuredSerializer<$name> {
   final String wireName = '${escapeString(wireName)}';
 
   @override
-  Iterable serialize(Serializers serializers, $name object,
+  Iterable<Object> serialize(Serializers serializers, $name object,
```
Comment (srawlins) [inline built_value_generator/lib/src/serializer_source_class.dart]: Er... did I do this?

Comment (davidmorgan) [reply built_value_generator/lib/src/serializer_source_class.dart]: Apparently. I think you're going to need that for the generated code to be raw type safe?

**Comment by davidmorgan** [comment]
Generally LG--let me know if you're happy with the `Iterable<Object>` part. I'm happy to merge and pull into google3; if you want it soon then please let me know, I'm OOO Friday :)

File: built_value_generator/lib/src/serializer_source_class.dart (hunk 2 above)
Comment (srawlins) [reply built_value_generator/lib/src/serializer_source_class.dart]: Haha yeah that's true. I'm just not sure how my change triggered it. But works for me!

**Comment by srawlins** [comment]
LG to me too. No rush before next week.

**Comment by srawlins** [comment]
I think Travis breakage is unrelated...

**Comment by davidmorgan** [comment]
Sorry for the slow response ... the travis breakage is from package:pedantic :D ... I'll take a look.

**Comment by davidmorgan** [comment]
Patched and reran presubmit after my PR to fix travis; LG; I'll rebase+merge... thanks!

**Pull request merged**